### DISCUSSION TITLE
macOS custom clients (arm and x64).  Key strokes not recognised (RustDesk Server Pro 1.4.5)

### DISCUSSION THREAD

**Discussion by PC-Support-Services** [rustdesk/rustdesk-server-pro#450] [Q&A]
I've had ongoing issues with using macOS custom clients not recognising key strokes no matter what keyboard mode (legacy, map or translate modes) I set or removing / re-adding input monitoring.

Has anyone managed to get these macOS custom clients to work or any idea how to fix the problem?

I've even fired up a UTM macOS VM (M1 processor) clean install of macOS and it doesn't work.

**Comment by rustdesk** [comment]
Please send your dmg to @21pages sunboeasy@gmail.com , he will test it for you.

**Reply by PC-Support-Services** [reply]
Both dmg's on their way

**Reply by 21pages** [reply]
Windows -> Macos, works on my computers.

https://github.com/user-attachments/assets/f209e018-80bb-43fa-9a99-332faefe0af6

**Reply by PC-Support-Services** [reply]
I do Mac (standard client0 --> Mac (custom client)

**Comment by rustdesk** [comment]
It does not work as controlled side or as controlling side?

**Reply by PC-Support-Services** [reply]
Controlled side

**Comment by rustdesk** [comment]
Please try out https://github.com/rustdesk/rustdesk/issues/3261#issuecomment-1768000251 

```
tccutil reset ScreenCapture com.carriez.RustDesk
tccutil reset Accessibility com.carriez.RustDesk
```


and https://github.com/rustdesk/rustdesk/discussions/7762#discussioncomment-9175161

```
bundleId='com.carriez.rustdesk'

sudo tccutil --service kTCCServiceSystemPolicyAllFiles --insert "${bundleId}"
sudo tccutil --service kTCCServiceSystemPolicyAllFiles --enable "${bundleId}"

sudo tccutil --service kTCCServiceScreenCapture --insert "${bundleId}"
sudo tccutil --service kTCCServiceScreenCapture --enable "${bundleId}"

sudo tccutil --service kTCCServiceListenEvent --insert "${bundleId}"
sudo tccutil --service kTCCServiceListenEvent --enable "${bundleId}"

sudo tccutil --service kTCCServiceAccessibility --insert "${bundleId}"
sudo tccutil --service kTCCServiceAccessibility --enable "${bundleId}"
```

#  Replace bundle id  with your custom bundle id

![image](https://github.com/user-attachments/assets/a84f6fb2-1c41-453f-a908-8fec1d27f6ad)

**Reply by PC-Support-Services** [reply]
Made no difference. 

tccutil: Failed to reset kTCCServiceSystemPolicyAllFiles approval status for com.xxxxx.rust

... other tccutil messages are the same

**Comment by PC-Support-Services** [comment]


**Reply by 21pages** [reply]
Ok, could you send another outgoing-able Mac client?
